.flake8

```diff
@@ -18,4 +19,5 @@ select = B,C,E,F,W,T,I,B
 ban-relative-imports = true
```

Comment:
ban-relative-imports is imported twice here. 

Run this script: 
```shell 
 make check-lintingc
``` 
The following errors will emerge from `mypy` which would be nice to fix as part of this PR.
```python
dynamicio/config/pydantic/io_resources.py:67: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
dynamicio/config/pydantic/io_resources.py:96: error: Argument "const" to "Field" has incompatible type "None"; expected "bool"  [arg-type]
dynamicio/mixins/with_s3.py:53: error: Incompatible default for argument "data" (default has type "None", argument has type "bytes")  [assignment]
dynamicio/mixins/with_s3.py:53: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
dynamicio/mixins/with_s3.py:53: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
Found 2 errors in 2 files (checked 19 source files)

```